inst/htmljs/ldavis.js

```diff
@@ -1343,15 +1369,99 @@ LDAvis = function(to_select, json_file) {
         }
 
         function state_reset() {
+            
+            // set the style of any clicked term back to be non-underline                    
+            var old_term_clicked_id = termID + vis_state.term_clicked;
+            if (vis_state.term_clicked != "") {
+                var oldterm = document.getElementById(old_term_clicked_id);
+                if (oldterm != null) {
+                    oldterm.style.textDecoration = null;  
+                }
+            }
+                
             if (vis_state.topic > 0) {
                 topic_off(document.getElementById(topicID + vis_state.topic));
+                // set the style of any topic clicked to be back to regular style
+                // (no thick border around topic circle)
+                var old_topic_clicked_id = topicID + vis_state.topic_clicked;
+                if (vis_state.topic_clicked > 0) {
+                    document.getElementById(old_topic_clicked_id).style.strokeWidth = null;
+                }
             }
             if (vis_state.term != "") {
                 term_off(document.getElementById(termID + vis_state.term));
             }
             vis_state.term = "";
             document.getElementById(topicID).value = vis_state.topic = 0;
             state_save(true);
+            
+            // make sure term ids are all correct
+            d3.selectAll(".terms").attr("id", function(d) {
+                return (termID + d.Term)
+            });
+            
+            // update shiny inputs to be null
+            if (inShinyMode) {
+                Shiny.onInputChange('ldavis_topic_clicked', null);
+                Shiny.onInputChange('ldavis_term_clicked', null);
+            }
+            
+            // set state of topic_clicked to 0, so we can click on topic x, reset
+            // vis, then click on topic x again without any problems
+            vis_state.topic_clicked = 0;
+        }
+        
+        function topic_click(newtopic, newtopic_num) {
+            if (!inShinyMode) {
+              return null;
+            }
+            // set style of clicked topic to have thicker border
+            newtopic.style.strokeWidth = 2;
+            
+            // set style of old selected topic back to regular border
+            var old_topic_clicked_id = topicID + vis_state.topic_clicked;
+            if (vis_state.topic_clicked > 0 && old_topic_clicked_id != this.id) {
+                document.getElementById(old_topic_clicked_id).style.strokeWidth = null;
+            }
+            
+            // save state of topic clicked
+            vis_state.topic_clicked = newtopic_num;
+            
+            // update shiny input$ldavis_topic_clicked object to be new topic clicked 
+            Shiny.onInputChange('ldavis_topic_clicked', newtopic_num);
+        
```

Comment:
Ideally, the input name should have information about the relevant outputId (so, on the server, you can listen/respond to events from a specific output). Ideally the API would work [like leaflet](https://rstudio.github.io/leaflet/shiny.html), but I'm not immediately sure how to implement, any ideas?

Reply:
I think this should be easy, as the outputId gets passed into the LDAvis code as a selector here:

https://github.com/cpsievert/LDAvis/blob/bdd7b01e7929dbe94df7aea14df48a4822d10c19/inst/shiny/shinyLDAvis.js#L13-L14

Note that we would need #39 to be merged to allow handling multiple ldavis outputs in the same app. 

I'll change the API as you suggested, and whenever #39 is merged than having the outputIds in the input names will come in handy.